core/services/vrf/v2/listener_v2_log_listener_test.go

```diff
@@ -0,0 +1,688 @@
+		mockChain.On("LogPoller").Return(th.LogPoller)
+	})
+
+	// Block 3 to finalityDepth. Ensure we have finality number of blocks
+	for i := 1; i < int(finalityDepth); i++ {
+		th.Client.Commit()
+	}
+
+	// Emit some logs in blocks finalityDepth+1 to finalityDepth+4.
+	numBlocks := 5
+	for i := 0; i < numBlocks; i++ {
+		_, err1 := th.Emitter.EmitLog1(th.Owner, []*big.Int{big.NewInt(int64(i))})
+		require.NoError(t, err1)
+		_, err1 = th.Emitter.EmitLog2(th.Owner, []*big.Int{big.NewInt(int64(i))})
+		require.NoError(t, err1)
+		th.Client.Commit()
+	}
+
+	// Blocks till now: 2 (in SetupTH) + 2 (empty blocks) + 5 (EmitLog blocks) = 9
+
+	// Calling Start() after RegisterFilter() simulates a node restart after job creation, should reload Filter from db.
+	require.NoError(t, th.LogPoller.Start(testutils.Context(t)))
+
+	// The poller starts on a new chain at latest-finality (finalityDepth + 5 in this case),
+	// Replaying from block 4 should guarantee we have block 4 immediately.  (We will also get
+	// block 3 once the backup poller runs, since it always starts 100 blocks behind.)
+	require.NoError(t, th.LogPoller.Replay(testutils.Context(t), 4))
+
+	// We should immediately have at least logs 4-7
+	logs, err := th.LogPoller.Logs(4, 7, EmitterABI.Events["Log1"].ID, th.EmitterAddress,
+		pg.WithParentCtx(testutils.Context(t)))
+	require.NoError(t, err)
+	require.Equal(t, 3, len(logs))
+
+	lastProcessedBlock, err := th.Listener.initializeLastProcessedBlock(th.Ctx)
+	require.Nil(t, err)
+	require.Equal(t, int64(6), lastProcessedBlock)
+}
+
+func TestInitProcessedBlock_NoUnfulfilledVRFReqs(t *testing.T) {
+	t.Parallel()
+
+	finalityDepth := int64(3)
+	th := SetupTH(t, false, finalityDepth, 3, 2, 1000, func(mockChain *evmmocks.Chain, curTH *TestHarness) {
+		mockChain.On("ID").Return(curTH.ChainID)
+		mockChain.On("LogPoller").Return(curTH.LogPoller)
+	})
+
+	// Block 3 to finalityDepth. Ensure we have finality number of blocks
+	for i := 1; i < int(finalityDepth); i++ {
+		th.Client.Commit()
+	}
+
+	// Create VRF request block and a fulfillment block
+	_, err2 := th.VRFLogEmitter.EmitRandomWordsRequested(th.Owner,
+		[32]byte(th.Listener.job.VRFSpec.PublicKey.MustHash().Bytes()),
+		big.NewInt(1), big.NewInt(105), 1, 10, 10000, 2, th.Owner.From)
+	require.NoError(t, err2)
+	th.Client.Commit()
+	_, err2 = th.VRFLogEmitter.EmitRandomWordsFulfilled(th.Owner, big.NewInt(1), big.NewInt(105), big.NewInt(10), true)
+	require.NoError(t, err2)
+	th.Client.Commit()
+
+	// Emit some logs in blocks to make the VRF req and fulfillment older than finalityDepth from latestBlock
+	numBlocks := 5
+	for i := 0; i < numBlocks; i++ {
+		_, err1 := th.Emitter.EmitLog1(th.Owner, []*big.Int{big.NewInt(int64(i))})
+		require.NoError(t, err1)
+		_, err1 = th.Emitter.EmitLog2(th.Owner, []*big.Int{big.NewInt(int64(i))})
+		require.NoError(t, err1)
+		th.Client.Commit()
+	}
+
+	// Blocks till now: 2 (in SetupTH) + 2 (empty blocks) + 2 (VRF req/resp block) + 5 (EmitLog blocks) = 11
```

Comment:
This is flaky test logic here, can we avoid having to count these sorts of things in the test? I recently fixed `TestStartHeartbeats` which had similar flaky logic.

Reply:
What is flaky in this test? The number of `th.Client.Commit`'s are deterministic. I added these comments for the reader to be able to easily modify the tests.

I thought if I compute the expected blockNumber with a complicated calculation from initialiseLastProcessedBlock, it'll be unclear to the reader and difficult to maintain it as well.